compiler/src/dotty/tools/dotc/printing/SyntaxHighlighting.scala

```diff
@@ -66,16 +60,44 @@ object SyntaxHighlighting {
       taken
     }
 
+    def isUpper(c: Char) = c.isUpper || c == '_' || c == '$'
+
+    def opChar(c:Char) = (c: @switch) match {
+      case '~' | '!' | '@' | '#' | '%' |
+           '^' | '*' | '+' | '-' | '<' |
+           '>' | '?' | ':' | '=' | '&' |
+           '|' | '\\' | '/' => true
+      case _ => false
+    }
+
+    def tryInterpolatorPrefix(n: Char, remaining: Stream[Char]): Option[String] = {
```

Comment:
I don't fully understand what you are doing below. Could you explain in detail?

The previous solution didn't work because it assumed that the interpolator prefix was on length 1.  So we just need to consume characters as long as they are valid prefixes. Something like:
```scala
val (prefix, after) = remaining.span(interpolationPrefixes.contains)
if (after.startsWith("\"")) {
   ...
```

Reply:
The problem was that it also didn't supported string literals with symbols in it. something like a_+""